langlib/lang.regexp/src/main/ballerina/regexp.bal

```diff
@@ -238,6 +310,16 @@ public type Replacement ReplacerFunction|string;
 
 # Replaces the first match of a regular expression.
 #
+# ```ballerina
+# string:RegExp r = re `0+`;
+#
+# r.replace("10010011", "*") ⇒ 1*10011
+#
+# r.replace("10010011", "*", 4) ⇒ 1001*11
+#
+# r.replace("122111", "*") ⇒ 122111
```

Comment:
Add an example for `replace` and `replaceAll` using replaceFunction

Reply:
[4ab39fc](https://github.com/ballerina-platform/ballerina-lang/pull/39096/commits/4ab39fca6922f71d8e27c8e43da56f502c86dd96)